test/unittest.cpp

```diff
@@ -24,18 +26,32 @@ bool TestMemoryLeaks() {
 	return test_memory_leaks;
 }
 
+bool SummarizeFailures() {
+	return summarize_failures;
+}
+
 } // namespace duckdb
 
 int main(int argc, char *argv[]) {
 	duckdb::unique_ptr<FileSystem> fs = FileSystem::CreateLocal();
 	string test_directory = DUCKDB_ROOT_DIRECTORY;
+	const char *filename("failures_summary.txt");
```

Comment:
I don't really understand why we are writing to a file now in the first place? It seems this file is only used internally (we write to it and then we read it again in the end). Can we keep this state in-memory instead?